examples/bfs_bidirectional.rs

```diff
@@ -0,0 +1,155 @@
+//! This example demonstrates the BFS bidirectional algorithm,
```

Comment:
Running this example on my machine

```
Corner to Corner
================
BFS took 5.821209ms
Bidirectional BFS took 7.437667ms

Center to Corner
================
BFS took 6.005333ms
Bidirectional BFS took 4.5025ms
```

The difference doesn't seem so big, but on higher dimension problems the difference can become very significant. E.g. think of the volume of 2 spheres of radius 1 vs the volume of 1 sphere of radius 2.